dev.py

```diff
@@ -409,7 +409,12 @@ def setup_build(args, env):
     """
     cmd = ["meson", "setup", args.build_dir, "--prefix", PATH_INSTALLED]
     build_dir = Path(args.build_dir)
-    if os.path.exists(build_dir / 'build.ninja'):
+    if (os.path.exists(build_dir) and not os.path.exists(build_dir
+                                                         / 'meson-info')):
+        if len(os.listdir(build_dir)):
```

Comment:
Similarly, though less idiomatically, maybe

```suggestion
        if list(build_dir.iterdir()):
```